standalone-metastore/src/main/java/org/apache/hadoop/hive/metastore/ReplChangeManager.java

```diff
@@ -163,104 +163,99 @@ public boolean accept(Path p){
    * @param ifPurge if the file should skip Trash when move/delete source file.
    *                This is referred only if type is MOVE.
    * @return int
-   * @throws MetaException
+   * @throws IOException
    */
-  int recycle(Path path, RecycleType type, boolean ifPurge) throws MetaException {
+  public int recycle(Path path, RecycleType type, boolean ifPurge) throws IOException {
```

Comment:
Why try-catch moved to callers?

Reply:
since acid Cleaner does not support the metaexception its moved to other places from the base method which throws compatible exception for both use cases.